src/controllers/user.routes.ts

```diff
@@ -0,0 +1,49 @@
+import { Router } from 'express';
+import { ValidatedRequest } from "express-joi-validation";
+import { UserModelAttributes } from '../interfaces/user.types';
+import { createUser, editUser, getUserById, getUsersByLogin, removeUser } from "../services/user.methods";
+import { ICreateUserSchema, IGetUserSchema, IUserSchema, createUserSchema, editUserSchema, getUserScheme, validator } from "../services/user.validation";
+
+const userPath = '/user/:id?:login?';
+
+export const userRouter = (router: Router, users: UserModelAttributes): void => {
+    router.route(userPath)
+        .post(validator.body(createUserSchema), async (req: ValidatedRequest<ICreateUserSchema>, res) => {
+            try {
+                const user = await createUser(req.body, users);
+
+                res.json(user)
+            } catch (e) {
+                res.end(e.message)
+            }
+        })
+        .get(validator.query(getUserScheme), async (req: ValidatedRequest<IGetUserSchema>, res) => {
+            try {
+                let user;
+
+                if (req.query.id) user = await getUserById(req.query.id, users);
+                else user = await getUsersByLogin(req.query.login, req.query.limit, users);
```

Comment:
What if you have both `login` and `id` in request?

Reply:
we will use id like a primary field, what the problem here?